File: CPlusPlus/Memory/src/GameRole.cpp
/*************************************************************************
    > File Name: GameRole.cpp
    > Author: ma6174
    > Mail: ma6174@163.com 
    > Created Time: 2017年03月23日 星期四 16时25分57秒
 ************************************************************************/

#include"GameRole.h"
#include<iostream>

using namespace std;

GameRole::GameRole():vitality(100), attack(100), defence(100)
{
	;
}

GameRole::~GameRole()
{
	;
}

Memento GameRole::Save()
{
	Memento mem(vitality, attack, defence);
	return mem;
}

void GameRole::Load(const Memento& mem)
{
	vitality=mem.GetVitality();
	attack=mem.GetAttack();
	defence=mem.GetDefence();
}

void GameRole::Show()
{
	cout<<"vitality:"<<vitality<<" "<<"attack:"<<attack<<" "<<"defence:"<<defence<<endl;
}

void GameRole::BeingAttacked()
{
	vitality-=10;
	attack-=10;
	defence-=10;
}

File: Python/Memento/GameRole.py
# -*- coding:utf-8 -*-

from Memento import *

class GameRole:

    def __init__(self):
        self.__vitality = 100
        self.__attack = 100
        self.__defence = 100
    
    def Save(self):
        mem = Memento(self.__vitality, self.__attack, self.__defence)
        return mem
    
    def Load(self, mem):
        self.__vitality = mem.GetVitality()
        self.__attack = mem.GetAttack()
        self.__defence = mem.GetDefence()

    def Show(self):
        print "%s%d%s%s%d%s%s%d"%("vitality:", self.__vitality, " ", "attack:", self.__attack, " ", "defence:", self.__defence)

    def BeingAttacked(self):
        self.__vitality -= 10
        self.__attack -= 10
        self.__defence -= 10
